magento/magento2
# magento/magento2#32309: IncosistentReferenceDefinition wrong check result

Re-creation of https://github.com/magento/magento2/pull/32624 as I was unable to push test coverage to original PR.



### Description (*)


### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#32309

### Manual testing scenarios (*)

1. Open db_schema.xml and change type for some column with a foreign key (for example, open `app/code/Magento/User/etc/db_schema.xml`, change `user_id` column type to `float`, and remove identity attribute).
2. Run `bin/magento setup:upgrade`.
3. Upgrade should fail with message `Column definition "some_column" and reference column definition "some_other_column" are different in tables "some_table" and "some_other_table"
`

### Questions or comments


### Contribution checklist (*)
